src/codegen/generate-node-errors.ts

```diff
@@ -0,0 +1,128 @@
+import NodeErrors from "../bun.js/bindings/ErrorCode.ts";
+const outputDir = process.argv[2];
+import path from "node:path";
+
+if (!outputDir) {
+  throw new Error("Missing output directory");
+}
+
+let enumHeader = ``;
+let listHeader = ``;
+let zig = ``;
+
+enumHeader = `
+// clang-format off
+// Generated by: src/codegen/generate-node-errors.ts
+#pragma once
+
+namespace Bun {
+  static constexpr size_t NODE_ERROR_COUNT = ${NodeErrors.length};
+  enum class ErrorCode : uint8_t {
+`;
+
+listHeader = `
+// clang-format off
+// Generated by: src/codegen/generate-node-errors.ts
+#pragma once
+
+struct ErrorCodeData {
+    JSC::ErrorType type;
+    WTF::ASCIILiteral name;
+    WTF::ASCIILiteral code;
+};
+static constexpr ErrorCodeData errors[${NodeErrors.length}] = {
+`;
+
+zig = `
+// Generated by: src/codegen/generate-node-errors.ts
+const std = @import("std");
+const bun = @import("root").bun;
+const JSC = bun.JSC;
+
+fn ErrorBuilder(comptime code_: Error, comptime fmt_: [:0]const u8, Args: type) type {
```

Comment:
these variables do not need to be named `code_` and `fmt_`, they can just be named `code` and `fmt`. not a big deal because generated code.